src/azure-cli/azure/cli/command_modules/aro/_aad.py

```diff
@@ -3,109 +3,55 @@
+        request_body = {"displayname": display_name, }
+        app = self.client.application_create(request_body)
+        obj_id = app["id"]
+        app_id = app["appId"]
+        password = self.add_password(obj_id)
+        return app_id, password
 
     def get_service_principal(self, app_id):
-        sps = list(self.client.service_principals.list(
-            filter="appId eq '%s'" % app_id))
+        sps = self.client.service_principal_list(f"appId eq '{app_id}'")
         if sps:
-            return sps[0]
+            return sps[0]["id"]
         return None
 
     def get_application_by_client_id(self, client_id):
-        apps = list(self.client.applications.list(
-            filter="appId eq '%s'" % client_id))
+        apps = self.client.application_list(f"appId eq '{client_id}'")
         if apps:
-            return apps[0]
+            return apps[0]["id"]
         return None
 
     def create_service_principal(self, app_id):
         max_retries = 3
         retries = 0
         while True:
             try:
-                return self.client.service_principals.create(
-                    ServicePrincipalCreateParameters(app_id=app_id))
-            except GraphErrorException as ex:
+                return self.client.service_principal_create({"appId": app_id})["id"]
+            except GraphError as ex:
                 if retries >= max_retries:
                     raise
                 retries += 1
                 logger.warning("%s; retry %d of %d", ex, retries, max_retries)
                 time.sleep(10)
 
-    def refresh_application_credentials(self, object_id):
-        password = uuid.uuid4()
-        key_id = uuid.uuid4()
-        start_date = datetime.datetime.utcnow()
-        end_date = datetime.datetime(2299, 12, 31, tzinfo=datetime.timezone.utc)
-
-        try:
-            credentials = list(self.client.applications.list_password_credentials(object_id))
-        except GraphErrorException as e:
-            logger.error(e.message)
-            raise
-
-        # keys created with older version of cli are not updatable
-        # https://github.com/Azure/azure-sdk-for-python/issues/18131
-        for c in credentials:
-            if c.custom_key_identifier is None:
-                raise BadRequestError("Cluster AAD application contains a client secret with an empty identifier.\n\
-Please either manually remove the existing client secret and run `az aro update --refresh-credentials`, \n\
-or manually create a new client secret and run `az aro update --client-secret <ClientSecret>`.")
-
-        # when appending credentials ALL fields must be present, otherwise
-        # azure gives ambiguous errors about not being able to update old keys
-        credentials.append(PasswordCredential(
-            custom_key_identifier=str(start_date).encode(),  # bytearray
-            key_id=str(key_id),
-            start_date=start_date,
-            end_date=end_date,
-            value=password))
```

Comment:
Glad to know `addPassword` no longer requires all fields to be present: https://docs.microsoft.com/en-us/graph/api/application-addpassword?view=graph-rest-1.0&tabs=http

Reply:
Yeah, that's right.  I tested it and it worked without these parameters.  In fact, you can no longer set the password and custom key identifier.  The end date is also always less than 2 years.